### ISSUE TITLE
Asyncio incompatibility: address already in use

### ISSUE THREAD

**Issue by Tinche** [MagicStack/uvloop#25]
Hello,

I thought I'd give the aiofiles tests a go on uvloop and had a few failing. Here's a minimal program that works under the default asyncio, and doesn't with uvloop:

```
import asyncio
import socket

from os.path import join, dirname

import uvloop

asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())

async def test_sendfile_socket():
    """Test the sendfile functionality, file-to-socket."""
    port = 50000

    @asyncio.coroutine
    def serve_file(_, writer):
        pass

    server = await asyncio.start_server(serve_file, port=port)

    await server.wait_closed()


loop = asyncio.get_event_loop()

loop.run_until_complete(test_sendfile_socket())
```

```
> python t.py
Traceback (most recent call last):
  File "uvloop/loop.pyx", line 1168, in uvloop.loop.Loop.create_server (uvloop/loop.c:19802)
  File "uvloop/handles/streamserver.pyx", line 38, in uvloop.loop.UVStreamServer.listen (uvloop/loop.c:56668)
  File "uvloop/handles/streamserver.pyx", line 75, in uvloop.loop.UVStreamServer._fatal_error (uvloop/loop.c:57149)
OSError: [Errno 98] Address already in use

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "t.py", line 25, in <module>
    loop.run_until_complete(test_sendfile_socket())
  File "uvloop/loop.pyx", line 1015, in uvloop.loop.Loop.run_until_complete (uvloop/loop.c:18124)
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/asyncio/tasks.py", line 240, in _step
    result = coro.send(None)
  File "t.py", line 18, in test_sendfile_socket
    server = await asyncio.start_server(serve_file, port=port)
  File "/usr/lib/python3.5/asyncio/streams.py", line 116, in start_server
    return (yield from loop.create_server(factory, host, port, **kwds))
  File "uvloop/loop.pyx", line 1172, in create_server (uvloop/loop.c:19927)
OSError: [Errno 98] error while attempting to bind on address ('::', 50000, 0, 0): address already in use
```

64-bit Ubuntu 16.04, Python 3.5. Using `reuse_port=True` will solve it, not sure if that's the difference.

**Comment by 1st1** [comment]
I don't think that anything is wrong here, it seems that you have something listening on port 50000. 

I've tried running this program and it started just fine. I also cloned aiofiles and modified `aiofiles/__init__.py` to install uvloop policy, all tests run fine (except `test_sendfile_file` which fails in the same way for asyncio and uvloop loops).

**Comment by Tinche** [comment]
Hm, weird. I definitely do not have anything running on 50000, and I tried other ports. 100% reproducible. Uvloop 0.4.23.

**Comment by 1st1** [comment]
Can you try different port, say 60000?

**Comment by Tinche** [comment]
```
> python t.py
Traceback (most recent call last):
  File "uvloop/loop.pyx", line 1168, in uvloop.loop.Loop.create_server (uvloop/loop.c:19802)
  File "uvloop/handles/streamserver.pyx", line 38, in uvloop.loop.UVStreamServer.listen (uvloop/loop.c:56668)
  File "uvloop/handles/streamserver.pyx", line 75, in uvloop.loop.UVStreamServer._fatal_error (uvloop/loop.c:57149)
OSError: [Errno 98] Address already in use

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "t.py", line 25, in <module>
    loop.run_until_complete(test_sendfile_socket())
  File "uvloop/loop.pyx", line 1015, in uvloop.loop.Loop.run_until_complete (uvloop/loop.c:18124)
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/asyncio/tasks.py", line 240, in _step
    result = coro.send(None)
  File "t.py", line 18, in test_sendfile_socket
    server = await asyncio.start_server(serve_file, port=port)
  File "/usr/lib/python3.5/asyncio/streams.py", line 116, in start_server
    return (yield from loop.create_server(factory, host, port, **kwds))
  File "uvloop/loop.pyx", line 1172, in create_server (uvloop/loop.c:19927)
OSError: [Errno 98] error while attempting to bind on address ('::', 60000, 0, 0): address already in use
```

**Comment by 1st1** [comment]
Interesting. And it doesn't fail if you comment out the uvloop-policy line?

What's your OS, Python version, how do you run the test program?

**Comment by 1st1** [comment]
Alright, I've reproduced this on my Linux box.

**Comment by Tinche** [comment]
Correct, it works if I comment out the policy switch.

Ubuntu 16.04, kernel 4.4.0-22-generic, Python 3.5.1+ virtualenv. pip list:
```
pip (8.1.1)
pkg-resources (0.0.0)
setuptools (20.7.0)
uvloop (0.4.23)
```
While virtualenv is active, just `python t.py`.

**Comment by 1st1** [comment]
Should be fixed in uvloop `v0.4.24` (see 76bb934d802f33aced864514c7ff9904df3d4d20 for details)!

**Comment by Tinche** [comment]
Confirm, my test suite passes now :)
